endaq/calc/filters.py

```diff
@@ -18,6 +18,16 @@ def butterworth(
 
     This function uses Butterworth filter designs, and implements the filter(s)
     as bi-directional digital biquad filters, split into second-order sections.
+
+    :param df: the input data; cutoff frequencies are relative to the
+        timestamps in `df.index`
+    :param low_cutoff: the low-frequency cutoff, if any; frequencies below this
+        value are rejected, and frequencies above this value are preserved
+    :param high_cutoff: the high-frequency cutoff, if any; frequencies above
+        this value are rejected, and frequencies below this value are preserved
+    :param half_order: half of the order of the filter; higher orders provide
+        more aggressive stopband reduction
+    :return: the filtered data
```

Comment:
Adding this here, but it's applicable in general.  Should we include docstring references to the `numpy`/`scipy` functions underlying out implementations?  We often get questions about how things are implemented, and giving users easy access would be reasonably straightforward.
```suggestion

    :param df: the input data; cutoff frequencies are relative to the
        timestamps in `df.index`
    :param low_cutoff: the low-frequency cutoff, if any; frequencies below this
        value are rejected, and frequencies above this value are preserved
    :param high_cutoff: the high-frequency cutoff, if any; frequencies above
        this value are rejected, and frequencies below this value are preserved
    :param half_order: half of the order of the filter; higher orders provide
        more aggressive stopband reduction
    :return: the filtered data
    
    .. seealso::
      
      `Scipy Butterworth filter design <https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.butter.html>`_
      Documentation for the function used for butterworth filter design.
```

Reply:
Based on the kinds of questions I've seen customers ask, and the sort of thing that might need clarification, I would say `scipy.integrate.cumulative_trapezoid` doesn't need to be referenced, "cumulative trapezoid method" really has all the details.  "We're using a Tukey window with these parameters" is also self explanatory, but more of what I would consider to be on the line.  On the other hand, the exact design of a butterworth filter could be (slightly) different between implementations, so referencing `scipy.signal.butter` and `scipy.signal.lfiltfilt` would help us out.